src/test/java/com/epam/tc/hw2/ex1/FirstExercise.java

```diff
@@ -0,0 +1,135 @@
+        softly.assertThat(actualTextUnderFirstImage).isEqualTo(EXPECTED_TEXT_UNDER_FIRST_IMAGE);
+
+        String actualTextUnderSecondImage = driver.findElement(TEXT_UNDER_SECOND_IMAGE).getText();
+        softly.assertThat(actualTextUnderSecondImage).isEqualTo(EXPECTED_TEXT_UNDER_SECOND_IMAGE);
+
+        String actualTextUnderThirdImage = driver.findElement(TEXT_UNDER_THIRD_IMAGE).getText();
+        softly.assertThat(actualTextUnderThirdImage).isEqualTo(EXPECTED_TEXT_UNDER_THIRD_IMAGE);
+
+        String actualTextUnderFourthImage = driver.findElement(TEXT_UNDER_FOURTH_IMAGE).getText();
+        softly.assertThat(actualTextUnderFourthImage).isEqualTo(EXPECTED_TEXT_UNDER_FOURTH_IMAGE);
+
+        //8. Assert that there is the iframe with “Frame Button” exist
+        boolean isActualIframeExists = (new WebDriverWait(driver, 5))
+                .until(ExpectedConditions.invisibilityOfElementWithText(IFRAME, "The iframe exists"));
+        softly.assertThat(isActualIframeExists).isTrue();
+
+        //9. Switch to the iframe and check that there is “Frame Button” in the iframe
+        WebElement iframeElement = driver.findElement(IFRAME);
+        driver.switchTo().frame(iframeElement);
+
+        boolean isFrameButtonElementPresent = (new WebDriverWait(driver, 5))
+                .until(ExpectedConditions
+                        .invisibilityOfElementWithText(FRAME_BUTTON, "The 'Frame Button' exists"));
+        softly.assertThat(isFrameButtonElementPresent).isTrue();
+
+        //10. Switch to original window back
+        driver.switchTo().defaultContent();
+
+        //11. Assert that there are 5 items in the Left Section are displayed and they have proper text
+        boolean isHomeItemDisplayed = (new WebDriverWait(driver, 5))
+                .until(ExpectedConditions.invisibilityOfElementWithText(HOME_ITEM, "Home item exists"));
+        softly.assertThat(isHomeItemDisplayed).isTrue();
+
+        boolean isContactFormItemDisplayed = (new WebDriverWait(driver, 5))
+                .until(ExpectedConditions.invisibilityOfElementWithText(CONTACT_FORM_ITEM, "Home item exists"));
+        softly.assertThat(isContactFormItemDisplayed).isTrue();
+
+        boolean isServiceItemDisplayed = (new WebDriverWait(driver, 5))
+                .until(ExpectedConditions.invisibilityOfElementWithText(SERVICE_ITEM, "Home item exists"));
+        softly.assertThat(isServiceItemDisplayed).isTrue();
+
+        boolean isMetalsAndColorsItemDisplayed = (new WebDriverWait(driver, 5))
+                .until(ExpectedConditions.invisibilityOfElementWithText(METALS_AND_COLORS_ITEM, "Home item exists"));
+        softly.assertThat(isMetalsAndColorsItemDisplayed).isTrue();
+
+        boolean isElementsPacksItemDisplayed = (new WebDriverWait(driver, 5))
+                .until(ExpectedConditions.invisibilityOfElementWithText(ELEMENTS_PACKS_ITEM, "Home item exists"));
+        softly.assertThat(isElementsPacksItemDisplayed).isTrue();
```

Comment:
please see comment for images

Reply:
I collected web elements with link to the List, converted it to the text using stream and compared actual and expected results.